src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -177,24 +256,35 @@ boot_comp_map_populate(struct cobj_header *h, spdid_t spdid, vaddr_t comp_info,
 
 	start_addr = (char *)(new_comp_cap_info[spdid].vaddr_mapped_in_booter);
 	init_daddr = cobj_sect_get(h, 0)->vaddr;
+
+	int total = 0;
 	for (i = 0; i < h->nsect; i++) {
 		struct cobj_sect *sect;
 		vaddr_t           dest_daddr;
 		char *            lsrc;
 		int               left;
-
+		
 		sect = cobj_sect_get(h, i);
 		/* virtual address in the destination address space */
 		dest_daddr = sect->vaddr;
 		/* where we're copying from in the cobj */
 		lsrc = cobj_sect_contents(h, i);
 		/* how much is left to copy? */
 		left = cobj_sect_size(h, i);
+		total += left;
 
 		/* Initialize memory. */
-		if (!(sect->flags & COBJ_SECT_KMEM) && (first_time || !(sect->flags & COBJ_SECT_INITONCE))) {
+		if (!(sect->flags & COBJ_SECT_KMEM)) {
 			if (sect->flags & COBJ_SECT_ZEROS) {
-				memset(start_addr + (dest_daddr - init_daddr), 0, left);
+				/* HACK FOR PING COMPONENT
+				 * For some reason ping memsets one more page than it is allowed to.
+				 */
+				if (left > PAGE_SIZE*257) {
```

Comment:
what is this?  someone has to debug this as this code isn't usable with any amount of a different setup.  Who should?